docs/source/cpp/examples/dataset_skyhook_scan_example.rst

```diff
@@ -0,0 +1,85 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+
+Arrow Skyhook example
+=========================
+
+The file ``cpp/examples/arrow/dataset_skyhook_scan_example.cc``
+located inside the source tree contains an example of using Skyhook to 
+offload filters and projections to a Ceph cluster.
+
+Instuctions
+--------------------
+
+1. Install Ceph and Skyhook dependencies.
+.. code-block:: bash
+
+    apt update 
+    apt install -y cmake \
+                    libradospp-dev \
+                    rados-objclass-dev \
+                    ceph \
+                    ceph-common \
+                    ceph-osd \
+                    ceph-mon \
+                    ceph-mgr \
+                    ceph-mds \
+                    rbd-mirror \
+                    ceph-fuse
+
+2. Build and install Skyhook.
+
+.. code-block:: bash
+
+    git clone https://github.com/apache/arrow
+    cd arrow/
+    mkdir -p cpp/release
+    cd cpp/release
+    cmake -DARROW_SKYHOOK=ON \
+    -DARROW_PARQUET=ON \
+    -DARROW_WITH_SNAPPY=ON \
+    -DARROW_BUILD_EXAMPLES=ON \
+    -DARROW_DATASET=ON \
+    -DARROW_CSV=ON \
+    -DARROW_WITH_LZ4=ON \
+    ..
```

Comment:
nit, but the indentation-after-backslash is a little inconsistent for each of these code blocks